annif/eval.py

```diff
@@ -145,10 +145,43 @@ def _evaluate_samples(self, y_true, y_pred, metrics='all'):
 
         return results
 
-    def results(self, metrics='all'):
+    def output_result_per_label(self, y_true, y_pred, results_file):
+        """Write results per label (non-aggregated) to outputfile"""
+
+        y_pred = y_pred.T > 0.0
+        y_true = y_true.T > 0.0
+
+        tp = (y_true & y_pred)
+        fp = (~y_true & y_pred)
+        fn = (y_true & ~y_pred)
+
+        r = len(y_true)
+
+        with open(results_file, 'w') as f:
+            f.write('\t'.join(['URI', 'Label', 'Support', 'True_positives',
+                               'False_positives', 'False_negatives',
+                               'Precision', 'Recall', 'F1_score']) + '\n')
+            zipped = zip(self._subject_index._uris,       # URI
+                         self._subject_index._labels,     # Label
+                         [sum(tp[i]) + sum(fn[i])
+                          for i in range(r)],             # Support
+                         [sum(tp[i]) for i in range(r)],  # True_positives
+                         [sum(fp[i]) for i in range(r)],  # False_positives
+                         [sum(fn[i]) for i in range(r)],  # False_negatives
+                         [precision_score(y_true[i], y_pred[i], zero_division=0)
+                          for i in range(r)],             # Precision
+                         [recall_score(y_true[i], y_pred[i], zero_division=0)
+                          for i in range(r)],             # Recall
+                         [f1_score(y_true[i], y_pred[i], zero_division=0)
+                          for i in range(r)])             # F1
+            f.write('\n'.join('\t'.join(str(e) for e in row)
```

Comment:
This is probably a matter of style, but I think it would be clearer to use a more traditional for loop and a print command, something like:

```python
for row in zipped:
    print('\t'.join((str(e) for e in row)), file=f)
```